src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -1861,96 +1861,95 @@ public ProtoCore.DSASM.ProcedureNode GetProcedureFromInstance(int classScope, Pr
         /// AST node
         /// </summary>
         /// <param name="node"></param>
-        /// <returns></returns>
+        /// <returns>Replication guide data</returns>
         private List<AssociativeNode> GetReplicationGuides(AssociativeNode node)
         {
-            
-            if (node is ArrayNameNode)
+            if (node is ArrayNameNode nameNode)
             {
-                var guides = ((ArrayNameNode) node).ReplicationGuides;
-                if (guides == null || guides.Count == 0)
+                var guides = nameNode.ReplicationGuides;
+                if ((guides == null || guides.Count == 0) && nameNode is IdentifierListNode identListNode)
                 {
-                    if (node is IdentifierListNode)
-                    {
-                        var identListNode = node as IdentifierListNode;
-                        return GetReplicationGuides(identListNode.RightNode);
-                    }
+                    return GetReplicationGuides(identListNode.RightNode);
                 }
                 return guides;
             }
             
-            if (node is FunctionDotCallNode)
+            if (node is FunctionDotCallNode dotCallNode)
             {
-                var dotCallNode = node as FunctionDotCallNode;
                 return GetReplicationGuides(dotCallNode.FunctionCall.Function);
             }
 
             return null;
         }
 
+        /// <summary>
+        /// This helper function extracts the at level data from the 
+        /// AST node
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns>At level data</returns>
         private AtLevelNode GetAtLevel(AssociativeNode node)
         {
-            var n1 = node as ArrayNameNode;
-            if (n1 != null)
+            if (node is IdentifierListNode identifierListNode)
```

Comment:
can you add a note (in the code) here that the order matters ?
First check the most derived class IdentifierListNode, then the base class ArrayNameNode